src/main/scala/com/fulcrumgenomics/bam/SamRecordClipper.scala

```diff
@@ -333,29 +333,31 @@ class SamRecordClipper(val mode: ClippingMode, val autoClipAttributes: Boolean)
     * @param rec the record to examine
     */
   def numBasesExtendingPastMate(rec: SamRecord): Int = {
-    val mateEnd = rec.mateEnd.getOrElse(rec.start + abs(rec.insertSize) - 1)
-    numBasesExtendingPastMate(rec=rec, mateEnd=mateEnd)
+    val mateUnclippedStart = rec.mateUnclippedStart.getOrElse(rec.mateStart)
+    val mateUnclippedEnd   = rec.mateUnclippedEnd.getOrElse(rec.start + abs(rec.insertSize) - 1)
```

Comment:
In both places in this PR where you access the unclipped mate start/end positions I think you want to not consider any hard clip sequences and _**only**_ consider soft clip sequences. E.g. we don't want a read's soft-clip sequence to extend past, and into, the mates hard clip sequence.